pkgs/top-level/all-packages.nix

```diff
@@ -22076,7 +22076,10 @@ with pkgs;
 
   cpustat = callPackage ../os-specific/linux/cpustat { };
 
-  cockroachdb = callPackage ../servers/sql/cockroachdb { };
+  inherit (import ../servers/sql/cockroachdb pkgs)
+    cockroachdb_20_2 cockroachdb_21_1 cockroachdb_21_2;
+
+  cockroachdb = cockroachdb_21_2;
```

Comment:
This is going a bit quick. Previously, the version was 20.1. Now we're deleting 20.1. and jumping 4 major versions. Can we maybe instead keep 20.1 at least for a few months, mark it as deprecated by adding `cockroachdb_20_1` to `pkgs/top-level/aliases.nix` in the right place? We can bump the default `cockroachdb` to 20.2, and then bump the default version once with every nixpkgs LTS or so, until we're roughly in sync again.

Just saying because otherwise this is a possibly breaking change that comes unannounced.

Reply:
My thought process here is that the current version (20.1) has already reached EOL. This means there will be no more updates and (according to their release policy) have stopped helping people move to newer versions. With this, I assumed the package was out of date enough that it would be okay to make a major version change while skipping some intermediate versions. I also skipped 20.2 because the maintenance period is already over (2020-11-10) which means there are no more patches for it and they are just helping people move to newer versions (until 2022-05-10).

If it is preferable to go the longer way with version updates, I am more than happy to do that. Though, I should deprecate 20.2 as well because it will no longer receive patches.

Reply:
I just realized it is expecting a throw alias and that you cannot use this to also build the package? It has to be one or the other? I would rather find a different way considering 20.2 is not longer receiving patches, but still officially supported. Can we do something in the generic builder that will throw a warning about the version no longer receiving patches?

Reply:
> If it is preferable to go the longer way with version updates

I don't know about an official guideline, but as a cockroach user I'd expect from a good distribution that it's possible to update without leaving out versions.

> I just realized it is expecting a throw alias and that you cannot use this to also build the package?

That seems right, I misunderstood that.

> Can we do something in the generic builder that will throw a warning about the version no longer receiving patches?

If you throw a warning in the generic builder it will give a CI failure. But maybe you can put a warning in `aliases.nix`?

Reply:
The problem with `aliases.nix` is that it will not allow me to build that version while the warning is there. If we give some kind of deprecation message, it needs to be in a way that still builds. 

The biggest problem is that there is a soft and hard deprecation from cockroach. The soft being "no more patches, but we still support it while you move" and the hard being "zero support". It would be nice to give some kind of message that you are building a soft deprecated version of cockroach. But I'm entirely unsure how to do that.

Reply:
> I don't know about an official guideline, but as a cockroach user I'd expect from a good distribution that it's possible to update without leaving out versions.

Going forward we can do a more sensible upgrade path. The biggest problem is that this has been left unmaintained for a while. So users should expect a large upgrade in one go.

Reply:
I'm not sure about the deprecation messages either. If its ok for you, let's go for a compromise, bump the `cockroachdb` version, but still keep the old version around until the next LTS, and add a deprecation message simply to the package description.

Reply:
The message can simply be "Deprecated upstream, will be removed next nixpkgs LTS" or so.